app/uk/gov/hmrc/transitmovementsrouter/controllers/MessageController.scala

```diff
@@ -82,7 +88,22 @@ class MessagesController @Inject() (
             error => Status(error.code.statusCode)(Json.toJson(error)),
             response => Created.withHeaders("X-Message-Id" -> response.messageId.value)
           )
+    }
 
+  private def submit(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    messageType: MessageType,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier) =
+    if (isRequestMessageType(messageType)) {
```

Comment:
Rather than doing this here like this, which works against the pattern above, would it not be better to create this function here (from memory so may need tweaking, and this should be in the controller, because it's a presentation problem really):

```scala
def filterRequestMessageType(messageType: MessageType): EitherT[Future, PresentationError, RequestMessageType] = messageType match {
    case t: RequestMessageType => EitherT.rightT(t)
    case _ => EitherT.leftT(PresentationError.badRequest("messageType is not valid for requests"))
}
```

and use that in the for comprehension?

```scala
(for {
    messageType <- MessageTypeHeaderExtractor.extract(request.headers).asPresentation
    requestMessageType <- filterRequestMessageType(messageType)
    submitted   <- submit(movementType, movementId, messageId, requestMessageType, request.body)
} yield ...
```

and ditch this `submit` method.

Also note that it's a bad request that's appropriate here, because this service has done nothing wrong, it's the calling service that has.